client/app/lib/components/table/MuiTableAdapter/MuiTableToolbar.tsx

```diff
@@ -20,7 +20,42 @@ const MuiTableToolbar = (props: ToolbarProps): JSX.Element | null => {
   const { t } = useTranslation();
 
   if (props.alternative?.when())
-    return <ToolbarContainer>{props.alternative.render()}</ToolbarContainer>;
+    return (
+      <ToolbarContainer>
+        <div className="flex w-full justify-between">
+          {props.alternative.keepNative && props.onSearchKeywordChange && (
+            <SearchField
+              className="mr-4 lg:mr-0 lg:w-1/2"
+              onChangeKeyword={props.onSearchKeywordChange}
+              placeholder={props.searchPlaceholder ?? t(translations.search)}
+              value={props.searchKeyword}
+            />
+          )}
+
+          <div className="flex items-center">
+            {props.alternative.render()}
+
+            {props.alternative.keepNative && (
+              <>
+                {props.buttons}
+
+                {props.onDownloadCsv && (
+                  <Tooltip
+                    title={
+                      props.csvDownloadLabel ?? t(translations.downloadAsCsv)
+                    }
+                  >
+                    <IconButton onClick={props.onDownloadCsv}>
+                      <Download />
+                    </IconButton>
+                  </Tooltip>
+                )}
+              </>
+            )}
+          </div>
+        </div>
```

Comment:
This custom code looks like it should go into props.alternative.render() instead of touching the toolbar code itself.

Reply:
indeed, the implementation of this Table is that if the `alternative` is being introduced, and the condition suffices (i.e. some rows are getting selected), then this `alternative` will be rendered instead of the `native` one

In this particular situation, we want to keep whatever toolbar is initially rendered (i.e. Search bar and also Download CSV button) and hence I merely extend it by allowing user to indicate if they want to include the initial toolbar ones or not

Reply:
This is the wrong way to extend a library, there is too much duplicated code.

Reply:
Check failing rspec, one of the search fails, likely due to change in MUI table.

Reply:
it's because of the flaky test. I re-ran it and now it's all successful